src/community/web-resources/src/main/java/org/geoserver/web/resources/PageResourceBrowser.java

```diff
@@ -0,0 +1,516 @@
+import org.apache.wicket.ajax.markup.html.AjaxLink;
+import org.apache.wicket.behavior.AttributeAppender;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.link.AbstractLink;
+import org.apache.wicket.markup.html.link.Link;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.request.handler.resource.ResourceStreamRequestHandler;
+import org.apache.wicket.request.resource.ContentDisposition;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.web.GeoServerSecuredPage;
+import org.geoserver.web.treeview.TreeNode;
+import org.geoserver.web.treeview.TreeView;
+import org.geoserver.web.wicket.GeoServerDialog;
+import org.geoserver.web.wicket.ParamResourceModel;
+import org.geoserver.web.wicket.GeoServerDialog.DialogDelegate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * The ResourceBrowser page.
+ * 
+ * @author Niels Charlier
+ *
+ */
+public class PageResourceBrowser extends GeoServerSecuredPage {
+    
+    private static final long serialVersionUID = 3979040405548783679L;
+        
+    /**
+     * Behaviour for disabled button
+     */
+    private static final AttributeAppender DISABLED_BEHAVIOR = new AttributeAppender("class", new Model<String>("disabled"), " ");
+
+    /**
+     * The extension that are recognised as simple text resources (and can be edited with simple text editor).
+     */
+    private static final String[] TEXTUAL_EXTENSIONS = new String[] {"txt", "properties", "info", "xml", "sld", "rst", "log"};
+        
+    /**
+     * The expanded states model.
+     */
+    protected final ResourceExpandedStates expandedStates = new ResourceExpandedStates();
+    
+    /**
+     * The clip board.
+     */
+    protected final ClipBoard clipBoard;
+          
+    public PageResourceBrowser() {
+        //create the root node
+        final ResourceNode rootNode = new ResourceNode(store().get(Paths.BASE), expandedStates);
+        rootNode.getExpanded().setObject(true);
+        
+        //create tree view and clip board
+        final TreeView<Resource> treeView = new TreeView<Resource>("treeview", rootNode);
+        clipBoard = new ClipBoard(treeView);
+        
+        //used for all pop-up dialogs.
+        final GeoServerDialog dialog = new GeoServerDialog("dialog");
+        dialog.setResizable(false);        
+        
+        //upload button
+        final AjaxLink<Void> btnUpload = new AjaxLink<Void>("upload") {
+            private static final long serialVersionUID = -6538820444407766106L;
+
+            @Override
+            public void onClick(AjaxRequestTarget target) {
+                dialog.setInitialHeight(225);
```

Comment:
I have an aversion to hardcoded numeric literals, but in this context, it is self-documenting.